db/rdbms/src/main/java/io/camunda/db/rdbms/write/service/ProcessInstanceWriter.java

```diff
@@ -50,9 +50,34 @@ public void finish(
     }
   }
 
+  public void createIncident(final long key) {
+    final boolean wasMerged = mergeToQueue(key, (o, b) -> b.numIncidents(o.numIncidents() + 1));
+
+    if (!wasMerged) {
+      executionQueue.executeInQueue(
+          new QueueItem(
+              ContextType.PROCESS_INSTANCE,
+              key,
+              "io.camunda.db.rdbms.sql.ProcessInstanceMapper.incrementIncidentCount",
+              key));
+    }
+  }
+
+  public void resolveIncident(final long key) {
+    final boolean wasMerged = mergeToQueue(key, (o, b) -> b.numIncidents(o.numIncidents() - 1));
```

Comment:
Like discussed, remove the BiFunction in copy functions and use a getter at the builder here.